components/permissions_gates/any_team_permission_gate/any_team_permission_gate.tsx

```diff
@@ -0,0 +1,43 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+
+export type Props = {
+
+    /**
+     * Permissions enough to pass the gate (binary OR)
+     */
+    permissions: string[];
+
+    /**
+     * Has permission
+     */
+    hasPermission: boolean;
+
+    /**
+     * Invert the permission (used for else)
+     */
+    invert: boolean;
+
+    /**
+     * Content protected by the permissions gate
+     */
+    children: React.ReactNode;
+}
+
+export default class AnyTeamPermissionGate extends React.PureComponent<Props> {
+    static defaultProps = {
+        invert: false,
+    }
+
+    render() {
+        if (this.props.hasPermission && !this.props.invert) {
+            return this.props.children;
+        }
+        if (!this.props.hasPermission && this.props.invert) {
+            return this.props.children;
+        }
```

Comment:
this is basically a XOR operation, we can simplify it like
```suggestion
        if (this.props.hasPermission !== this.props.invert) {
            return this.props.children;
        }
```

Reply:
Thanks ! I committed your suggestion.